instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Includes work from:
+/*
+ * Copyright 2017-2021 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.opentelemetry.instrumentation.jdbc;
+
+import io.opentelemetry.instrumentation.jdbc.internal.DbInfo;
+import io.opentelemetry.instrumentation.jdbc.internal.JdbcConnectionUrlParser;
+import io.opentelemetry.instrumentation.jdbc.internal.JdbcMaps;
+import io.opentelemetry.instrumentation.jdbc.internal.OpenTelemetryConnection;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.DriverPropertyInfo;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+import java.util.logging.Logger;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class OpenTelemetryDriver implements Driver {
```

Comment:
Consider making it final and adding a private constructor

Reply:
Tried. But it doesn't work with HikariCP:

```
Caused by: java.lang.RuntimeException: Failed to load driver class io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver in either of HikariConfig class loader or Thread context classloader
	at com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:486)
	at io.micronaut.configuration.jdbc.hikari.$DatasourceConfigurationDefinition.injectBean(Unknown Source)
	at io.micronaut.configuration.jdbc.hikari.$DatasourceConfigurationDefinition.doBuild(Unknown Source)
	at io.micronaut.context.AbstractParametrizedBeanDefinition.build(AbstractParametrizedBeanDefinition.java:118)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:149)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1979)
	... 21 common frames omitted
```

Reply:
Sorry, not this one. Can make it final, but can not add a private constructor because of this exception:

```
Caused by: java.lang.RuntimeException: Failed to instantiate class io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver
	at com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:494)
Caused by: java.lang.RuntimeException: Failed to instantiate class io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver

	at io.micronaut.configuration.jdbc.hikari.$DatasourceConfigurationDefinition.injectBean(Unknown Source)
	at io.micronaut.configuration.jdbc.hikari.$DatasourceConfigurationDefinition.doBuild(Unknown Source)
	at io.micronaut.context.AbstractParametrizedBeanDefinition.build(AbstractParametrizedBeanDefinition.java:118)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:149)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1979)
	... 21 common frames omitted
Caused by: java.lang.NoSuchMethodException: io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3517)
Caused by: java.lang.NoSuchMethodException: io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver.<init>()

	at java.base/java.lang.Class.getConstructor(Class.java:2238)
	at com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:490)
	... 26 common frames omitted
```

Reply:
Okay, let's leave it like it is then. Thanks for checking this!